### ISSUE TITLE
unsollicited debug output "Level 0" upon PEP with on clause

### ISSUE THREAD

**Issue by tsmeele** [irods/irods#3500]
(irods 4.1.8, default resource = demoResc) 
Debug output is
```
-bash-4.2$ iput core.re c4
Level 0: DEBUG: 

Level 1: DEBUG: 

Level 2: DEBUG: 

-bash-4.2$ 
```
this happens when a pep_resource_modified_post exists that has an on clause which does **not** match, e.g.:
```
pep_resource_modified_post(*out) {
  on($pluginInstanceName == "irodsResc") {
    uuReplicateAsynchronously($KVPairs.logical_path, "irodsResc", "replResc");
#  on (true) { nop; }
}
```
The debug messages disappear when we add a default on clause (see commented out code) yet this would prohibit other existing rules to be tried.
Note: other policies might have similar issue, we simply found and tested it with this pep.

**Comment by alanking** [comment]
I cannot reproduce this issue in main.

...but did notice a few things...

The reference rule above is outdated/incorrect in a couple of ways:

1. It is missing a closing curly brace (assuming that's not a typo). Even when I resolved this, no debug output appeared for me.
2. The signature for dynamic PEPs seems to have changed since this issue was written (4.1.8). I changed it to include the instance and context parameters in the signature and even still no debug output appeared for me.

When I resolved these issues, I got this output:
```
remote addresses: 172.31.0.3 ERROR: putUtil: put error for /tempZone/home/alice/2022-05-09Z14:45:27--irods-testing-yraudedz/test_unsolicited_debug_messages_do_not_appear__issue_3500, status = -1101000 status = -1101000 RULE_FAILED_ERR
```

I changed the `on` clause to an `if` and everything became calm again. However, I found that I was not able to get anything to print to the server log. Turns out that `$pluginInstanceName` is an empty value for my `iput` operation.
